Plugins/Power Manager/Alerts & Reports/Python/get_power_manager_alerts.py

```diff
@@ -0,0 +1,207 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+SYNOPSIS:
+   Script to get the list of Power Manager Specific Alerts in OpenManage Enterprise
+
+DESCRIPTION:
+   This script exercises the OpenManage Enterprise REST API to get the list of Power Manager specific alerts
+    - For authentication, X-Auth is used over Basic Authentication
+    - Note that the credentials entered are not stored to disk.
+
+EXAMPLE:
+   python get_power_manager_alerts.py --ip <xx> --username <username> --password <pwd>
+   
+    Output:
+
+        =======================================
+            Power Manager - Metrics - Alerts
+        =======================================
+
+        SEVERITY  SOURCE_NAME      TIME                     CATEGORY       SUB_CATEGORY  MESSAGE_ID  MESSAGE
+
+        Warning   linux-0j8n       2020-03-20 16:31:20.484  System Health  Metrics       CMET0004    POWER on Group_R740s has exceeded its threshold.
+        Critical  linux-0j8n       2020-03-20 16:16:21.252  System Health  Metrics       CMET0008    TEMPERATURE on Group_R740s has exceeded its lower threshold.
+
+        ========================================================
+            Power Manager - Power Configuration - Alerts
+        ========================================================
+
+        SEVERITY  SOURCE_NAME      TIME                     CATEGORY       SUB_CATEGORY         MESSAGE_ID  MESSAGE
+
+        Normal    linux-0j8n       2020-01-23 13:30:00.399  System Health  Power Configuration  CPWR0014    Violation of power policy Policy_on_R740s on group Group_R740s got rectified.
+        Critical  linux-0j8n       2020-01-23 12:30:00.417  System Health  Power Configuration  CPWR0013    Power policy Policy_on_R740s on group Group_R740s got violated.
+
+"""
+
+#Import the modules required for this script
+import sys
+import argparse
+from argparse import RawTextHelpFormatter
+import json
+import requests
+import urllib3
+from columnar import columnar
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+
+def get_power_manager_alerts(ip_address, user_name, password):
+    """ Authenticate with OpenManage Enterprise, enumerate power manager alerts"""
+    try:
+        
+        # Defining Session URL & its headers
+        session_url = 'https://%s/api/SessionService/Sessions' % (ip_address)
+        headers = {'content-type': 'application/json'}
+        
+        # Define Payload for posting session API
+        user_details = {'UserName': user_name,
+                        'Password': password,
+                        'SessionType': 'API'}
+        
+        # Define the power manager alerts URL
+        alerts_metrics_url = "https://%s/api/AlertService/Alerts?$top=10000000&$filter=contains(SubCategoryName,'Metrics')" % (ip_address)
```

Comment:
We don't advocate using these semantics to retrieve all data - the right way is to handle the odata.nextlink param and iterate the call until the odata.nextLink is absent